app/scripts/metamask-controller.js

```diff
@@ -497,6 +515,29 @@ module.exports = class MetamaskController extends EventEmitter {
     }
   }
 
+  /**
+   * Get an account balance from the AccountTracker or request it directly from the network.
+   * @param {string} account - The account address
+   */
+  getBalance (account) {
+    return new Promise((resolve, reject) => {
+      const cached = this.accountTracker.store.getState().accounts[account]
+
+      if (cached && cached.balance) {
+        resolve(cached.balance)
+      } else {
```

Comment:
Unfortunately no... the accountTracker is not necessarily up to date at this point in the code. Many times it does not yet have a `balance`.

Reply:
Hmm, what I was getting at is that if we add a return statement after resolving the promise we don't need to have the rest of the function inside an else block since there isn't anything after this conditional.

It's certainly not a blocking comment though, more a small code-flow thing.